### PR TITLE
Utility classes for config and counters

### PR DESCRIPTION
**Pull request by ajaits**
Adding utlity classes
- config_map.py::ConfigMap: to load JSON dicts
- counters.py::Counters:

### CODE REVIEW

**Review by pradh — COMMENTED** [review]
Useful utils!

File: util/config_map.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''Class to store configuration parameters as a dictionary.
+
+Creates global flags:
+--config_file
+--config_params
```
[the remark this answers is not in the document]
Comment (ajaits) [reply util/config_map.py:18]: Yes, need the caller to pass these as arguments.

`_FLAGS(sys.argv)` to parse flags in the library before main() works locally, but breaks cloud build.

So better for caller to define the flags and pass them in. Added the flags to the sample usage comment.

File: util/config_map.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+Usage:
+   # Load config map from a json file with comments and trailing ,:
```
[the remark this answers is not in the document]
Comment (ajaits) [reply util/config_map.py:21]: Added a comment to the file header.

File: util/config_map.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+   # To load from command-line flags, set the following:
+   #  python3 ... \
+   #   --config_file=<my-config-file.json \
+   #   --config_params={'param1': <new-value> }'
+   #
+   from absl import flags
+
+   flags.DEFINE_string('config_file', '', 'File with configuration parameters.')
+   flags.DEFINE_string('config_params', '', 'Python dictionary as a string ')
```
[the remark this answers is not in the document]
Comment (ajaits) [reply util/config_map.py:41]: This is just an example to override a single or a few parameters from config file.

File: util/config_map.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            src[k].extend(v)
+        elif isinstance(v, set):
+            # TODO: deep update of set
+            src[k].update(v)
+        else:
+            src[k] = v
+    return src
+
+
+def get_py_dict_from_file(filename: str) -> dict:
```
[the remark this answers is not in the document]
Comment (ajaits) [reply util/config_map.py:87]: Added '_' prefix and moved to the end.

File: util/config_map.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+        Returns:
+            dictionary with all parameter:value mappings.
+        '''
+        if configs:
+            self._config_dict.update(configs)
+        return self._config_dict
+
+    def update_config(self, configs: dict) -> dict:
+        '''Does a deep update of the dict updating nested dicts as well.
```
[the remark this answers is not in the document]
Comment (ajaits) [reply util/config_map.py:175]: Added an example

File: util/counters.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        'debug': False,
+
+        # emit counters once every 30 secs.
+        'show_every_n_sec': 30,
+
+        # counter for records processed, used for computing processing rate.
+        'processed_counter': 'inputs',
+        # counter with total inputs, used for computing remaining time.
+        'total_counter': 'total',
+    }
```
[the remark this answers is not in the document]
Comment (ajaits) [reply util/counters.py:49]: Changed dict to NamedTuple. If there is a better alternative for that, pls let me know.

File: util/config_map.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajaits) [reply util/config_map.py]: Clarified in the doc string that this is optional

**Review by pulkit-s — APPROVED** [review]

**Pull request merged**